bridges/CssSelector2Bridge.php

```diff
@@ -0,0 +1,421 @@
+                // No `a` element found in this entry
+                if ($entry->tag == 'a') {
+                    $url_element = $entry;
+                } else {
+                    continue;
+                }
+            }
+
+            $links_with_elements[$url_element->href] = $entry;
+        }
+
+        if (empty($links_with_elements)) {
+            returnClientError('The provided URL selector matches some elements, but they do not 
+                contain links.');
+        }
+
+        // Filter using the URL pattern
+        $filtered_urls = $this->filterUrlList(array_keys($links_with_elements), $url_pattern, $limit);
+
+        if (empty($filtered_urls)) {
+            returnClientError('No results for URL pattern');
+        }
+
+        $items = [];
+        foreach ($filtered_urls as $link) {
+            $items[$link] = $links_with_elements[$link];
+        }
+
+        return $items;
+    }
+
+
+    /**
+     * Retrieve article element from its URL using content selector and return the DOM element
+     * @param string $entry_url URL to retrieve article from
+     * @param string $content_selector HTML selector for extracting content, e.g. "article.content"
+     * @return article DOM element
+     */
+    protected function fetchArticleElementFromPage($entry_url, $content_selector)
+    {
+        $entry_html = getSimpleHTMLDOMCached($entry_url);
+        $article_content = $entry_html->find($content_selector, 0);
+
+        if (is_null($article_content)) {
+            returnClientError('Could not article content at URL: ' . $entry_url);
+        }
+
+        $article_content = defaultLinkTo($article_content, $entry_url);
+        return $article_content;
+    }
+
+    /**
+     * Retrieve article content from its URL using content selector and return a feed item
+     * @param object $entry_html A DOM element containing the article
+     * @param string $title_selector A selector to the article title from the article
+     * @param string $author_selector A selector to find the article author
+     * @param string $time_selector A selector to get the article publication time.
+     * @param string $content_cleanup Optional selector for removing elements, e.g. "div.ads,
+     *  div.comments"
+     * @param string $title_default Optional title to use when could not extract title reliably
+     * @param bool $remove_styling Whether to remove class and style attributes from the HTML
+     * @return array Entry data: uri, title, content
+     */
+    protected function parseEntryElement(
+        $entry_html,
+        $title_selector = null,
+        $author_selector = null,
+        $category_selector = null,
+        $time_selector = null,
+        $content_cleanup = null,
+        $title_default = null,
+        $remove_styling = false,
+    ) {
+        $article_content = convertLazyLoading($entry_html);
+
+        if (is_null($title_selector)) {
+            $article_title = $title_default;
```

Comment:
I'd suggest using `$this->getTitle()` as fallback instead of `$this->feedName` when there is no title selector.